table/block_based_table_reader.cc

```diff
@@ -393,7 +398,10 @@ class BinarySearchIndexReader : public IndexReader {
   virtual InternalIterator* NewIterator(BlockIter* iter = nullptr,
                                         bool /*dont_care*/ = true,
                                         bool /*dont_care*/ = true) override {
-    return index_block_->NewIterator(icomparator_, iter, true);
+    return index_block_->NewIterator(
+        index_key_includes_seq_ ? icomparator_
+                                : icomparator_->user_comparator(),
```

Comment:
We never made Block::comparator_ typed InternalKeyComparator, which was an optimization opportunity. Now this change makes the optimization twice. Can we divide Block::comparator_ to Block:user_comparator_ and Block::internal_comparator_, and pick the one to use based on index_key_includes_seq_ ? The same branching can be effective to both of key_ and and which comparator to use. And hopefully they can be done inside Compare().

Reply:
Are you hoping that this would reduce the branch mis-prediction penalty?

Reply:
"Now this change makes the optimization twice" -> "Now this change makes the optimization impossible"...